x/icacallbacks/module_ibc.go

```diff
@@ -92,7 +94,11 @@ func (im IBCModule) NegotiateAppVersion(
 }
 
 // GetTxMsgData returns the msgs from an ICA transaction and can be reused across authentication modules
-func GetTxMsgData(ctx sdk.Context, ack channeltypes.Acknowledgement, logger log.Logger) (*sdk.TxMsgData, error) {
+func GetTxMsgData(ctx sdk.Context, ack channeltypes.Acknowledgement, logger log.Logger) (*types.ICATxResponse, error) {
```

Comment:
If you're on board with moving the ack unmarshaling out, I think we should add it to the top of this function and rename it to something more generic like `UnpackICAAcknowledgement`(I'm sure you can think of a much better name)

And then each callback would take types.ICATxResponse as the argument instead of the Ack packet type